src/metabase/driver/druid/query_processor.clj

```diff
@@ -100,42 +103,112 @@
 
 (declare filter:not filter:nil?)
 
+(defn- field? [arg]
+  (or (instance? Field arg)
+      (instance? DateTimeField arg)))
+
+(defn- expression->field-names [{:keys [args]}]
+  {:post [(every? u/string-or-keyword? %)]}
+  (flatten (for [arg   args
+                 :when (or (field? arg)
+                           (instance? Expression arg))]
+             (cond
+               (instance? Expression arg) (expression->field-names arg)
+               (field? arg)               (->rvalue arg)))))
+
+(defn- expression-arg->js [arg default-value]
+  (if-not (field? arg)
+    arg
+    (js/or (js/parse-float (->rvalue arg))
+           default-value)))
+
```

Comment:
`0` is falsely so it will always be replaced with the `default-value`, is that intended
?